### PR TITLE
Make Firebase Function code more readable

### PR DESCRIPTION
**Pull request by Matthias-Schaub**
Improve readability of the code.

### CODE REVIEW

File: firebase/functions/index.js [hunk 1]
```diff
@@ -129,10 +119,10 @@ exports.counter = functions.database.ref('/v2/results/{projectId}/{groupId}/{use
     return Promise.all(promises)
 })
 
-
-exports.counter_2 = functions.database.ref('/v2/users/{userId}/contributions/{projectId}/').onCreate((snapshot, context) => {
+// Counters to keep track of contributors and project contributions of Project and User.
+// Gets triggered when User contributes to new project.
+exports.contributionCounter = functions.database.ref('/v2/users/{userId}/contributions/{projectId}/').onCreate((snapshot, context) => {
```
Comment (Hagellach37) [inline firebase/functions/index.js:124]: can we change this to `contributorCounter` instead?

File: firebase/functions/index.js [hunk 2]
```diff
@@ -5,84 +5,70 @@ const functions = require('firebase-functions')
...
-
-    const taskContributionCountRef          = admin.database().ref('/v2/users/'+context.params.userId+'/taskContributionCount')
-    const groupContributionCountRef         = admin.database().ref('/v2/users/'+context.params.userId+'/groupContributionCount')
-    const projectContributionsRef           = admin.database().ref('/v2/users/'+context.params.userId+'/contributions/'+context.params.projectId)
-    const groupContributionsRef             = admin.database().ref('/v2/users/'+context.params.userId+'/contributions/'+context.params.projectId +'/'+context.params.groupId)
-    const projectTaskContributionCountRef   = admin.database().ref('/v2/users/'+context.params.userId+'/contributions/'+context.params.projectId+'/taskContributionCount')
-    const projectGroupContributionCountRef  = admin.database().ref('/v2/users/'+context.params.userId+'/contributions/'+context.params.projectId+'/groupContributionCount')
-    const totalTimeSpentMappingRef          = admin.database().ref('/v2/users/'+context.params.userId+'/timeSpentMapping')
+    const groupRef = admin.database().ref('/v2/groups/' + context.params.projectId + '/' + context.params.groupId)
```
Comment (Hagellach37) [inline firebase/functions/index.js:31]: nice, I like that you made this part simler.

File: firebase/functions/index.js [hunk 3]
```diff
@@ -5,84 +5,70 @@ const functions = require('firebase-functions')
...
+            return userRef.child('taskContributionCount').transaction((currentCount) => {
                 return currentCount + numberOfTasks
             })
         })
-    promises.push(taskContributionCount)
+    promises.push(totalTaskContributionCount)
 
-    // Counters for tasks and groups per user and per project
-    const projectGroupContributionCount = projectGroupContributionCountRef.transaction((currentCount) => {
+    // Counter for User per Project
```
Comment (Hagellach37) [inline firebase/functions/index.js:53]: I think this comment doesn't align to what happens in the function

File: firebase/functions/index.js [hunk 4]
```diff
@@ -5,84 +5,70 @@ const functions = require('firebase-functions')
...
 
-    const projectTaskContributionCount = numberOfTasksRef.once('value')
+    const taskContributionCount = groupRef.child('numberOfTasks').once('value')
         .then((dataSnapshot) => {
             const numberOfTasks = dataSnapshot.val()
             return numberOfTasks
         })
         .then((numberOfTasks) => {
-            return projectTaskContributionCountRef.transaction((currentCount) => {
+            return userContributionRef.child('taskContributionCount').transaction((currentCount) => {
```
Comment (Hagellach37) [inline firebase/functions/index.js:65]: this looks duplicated. What's the difference to the code in line 41 above?

Comment (Hagellach37) [reply firebase/functions/index.js:65]: probably here you wanted to add the number of tasks done for a specific project?

Comment (Hagellach37) [reply firebase/functions/index.js:65]: ah or it's my mistake and I didn't check the difference between `UserRef` and `UserContributionRef`...

**Review by Hagellach37 — APPROVED** [review]

File: firebase/functions/index.js (hunk 1 above)
Comment (Matthias-Schaub) [reply firebase/functions/index.js:124]: Renamed the function.

File: firebase/functions/index.js (hunk 3 above)
Comment (Matthias-Schaub) [reply firebase/functions/index.js]: Clarified comments.

**Pull request merged**